libsolidity/ast/AST.cpp

```diff
@@ -480,6 +480,66 @@ bool VariableDeclaration::canHaveAutoType() const
 	return isLocalVariable() && !isCallableParameter();
 }
 
+bool VariableDeclaration::isEventParameter() const
+{
+	return dynamic_cast<EventDefinition const*>(scope()) != nullptr;
+}
+
+bool VariableDeclaration::isReferenceType() const
+{
+	return dynamic_cast<ReferenceType const*>(typeName()->annotation().type.get());
```

Comment:
Please add `solAssert(typeName()->annotation().type, "Can only be called after reference resolution");`

Reply:
Also add this to the header file comment: Can only be called after reference resolution.